clients/src/test/java/org/apache/kafka/common/metrics/MetricsTest.java

```diff
@@ -202,6 +202,18 @@ public void testBadSensorHierarchy() {
         metrics.sensor("gc", c1, c2); // should fail
     }
 
+    @Test
+    public void testRemoveChildSensor() {
+        final Metrics metrics = new Metrics();
+
+        final Sensor parent = metrics.sensor("parent");
+        metrics.sensor("child", parent);
+
```

Comment:
Maybe worth a sanity assertion before removal of the child sensor? Something like this:
```java
assertTrue(metrics.childrenSensors().get(parent).contains(child));
```

Reply:
good idea.